### ISSUE TITLE
When polkit is missing, unprivileged calls fail with ServiceUnknown rather than AccessDenied

### ISSUE THREAD

**Issue by mvduin** [systemd/systemd#12209]
systemd v241 on debian buster

When systemd is compiled with PolicyKit support (as is the case on debian) but PolicyKit is installed, any dbus calls to systemd that require authorization fail with the wrong error:
```
~$ dbus-send --system --dest=org.freedesktop.systemd1 --print-reply /org/freedesktop/systemd1 org.freedesktop.systemd1.Manager.Reload
Error org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.PolicyKit1 was not provided by any .service files
```
The expected error is of course AccessDenied.

Based on a brief glance at the code in `shared/bus-util.c`, I see that there is detection code that is meant to intercept ServiceUnknown from polkit and replace it with AccessDenied (-EACCES), but evidently it's not working.

**Comment by mvduin** [comment]
Hmm, I think the problem is that `bus_verify_polkit_async` merely adjusts the return value to -EACCES but still copies the original error to the caller, unlike the synchronous `bus_test_polkit` which explicitly clears the error before returning -EACCES to ensure the error name and message will be generated from the return code.

**Comment by yuwata** [comment]
Fix is waiting in #12215.
